commit dd8636a93d91f430b09337667c2d6da6f87b383d (refs/changes/04/3604/3)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2018-03-13T21:01:56+00:00 (1 year, 7 months ago)
    
    TRITON-196 cloudapi machine objects don't show "deletion_protection: false" if it isn't set
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/machines.js b/lib/machines.js
index a97b05c..96df942 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -151,6 +151,7 @@ function translate(machine, req)  {
         ips: [],
         memory: Number(machine.ram),
         disk: (Number(machine.quota) * 1024) || 0,
+        deletion_protection: !!machine.indestructible_zoneroot,
         metadata: machine.customer_metadata || {},
         tags: machine.tags,
         credentials: credentials,
@@ -175,10 +176,6 @@ function translate(machine, req)  {
         msg.disk = Number(machine.disks[1].size);
     }
 
-    if (machine.indestructible_zoneroot) {
-        msg.deletion_protection = true;
-    }
-
     if (machine.nics && machine.nics.length) {
         machine.nics.forEach(function (nic) {
             msg.ips.push(nic.ip);
